query/expr/binaryscalarexpr.go

```diff
@@ -80,15 +84,30 @@ var ErrUnsupportedBinaryOperation = errors.New("unsupported binary operation")
 // operator may be satisfied by at least one value in the column chunk. If it
 // returns false, it means that the operator will definitely not be satisfied
 // by any value in the column chunk.
-func BinaryScalarOperation(left parquet.ColumnChunk, right parquet.Value, operator logicalplan.Op) (bool, error) {
+func BinaryScalarOperation(left parquet.ColumnChunk, right scalar.Scalar, operator logicalplan.Op) (bool, error) {
 	leftColumnIndex, err := left.ColumnIndex()
 	if err != nil {
 		return true, err
 	}
 	numNulls := NullCount(leftColumnIndex)
 	fullOfNulls := numNulls == left.NumValues()
+	leftColumnMax := Max(leftColumnIndex)
+	leftColumnMin := Min(leftColumnIndex)
+
+	// Because parquet doesn't support some types that arrow does, we need to
+	// convert the parquet min/max values to arrow scalars to be able to compare,
+	// using the right scalar as the type reference.
+	leftColumnMaxScalar, err := pqarrow.ParquetValueToArrowScalar(leftColumnMax, right.DataType())
```

Comment:
This is the most important change of this PR.
Given that we have the right side's DataType, we know exactly what type we want to compare it to. Using that information, we can be more precise when converting parquet to arrow.

Reply:
Hmm it feels bad to need to use the index to determine the scalar especially since the index is not a thing that is requried to exist.

Reply:
I think it's the other way around. We use the scalar passed into the binary scalar operation to determine what type the max and min types from the index should have.